sqlmesh/core/context.py

```diff
@@ -2018,6 +2020,12 @@ def _snapshot_gateways(self) -> t.Dict[str, str]:
             if snapshot.is_model and snapshot.model.gateway
         }
 
+    @cached_property
+    def engine_adapters(self) -> t.Dict[str, EngineAdapter]:
+        """Returns all engine adapters for the gateways defined in the configs."""
+        self._create_engine_adapters()
+        return self._engine_adapters
```

Comment:
Should we move the logic in `_create_engine_adapters` here?

cc @izeigerman does this look safe to cache?

Reply:
I considered this, only thing was that `_create_engine_adapters` takes the `gateways` as an argument and for the snapshot_evaluator we were only creating the ones used in the project rather than all.

Reply:
it was only used in the `snapshot_evaluator` before, when initialising the SnapshotEvaluator